### ISSUE TITLE
pcp integration does not work on armhf based platform

### ISSUE THREAD

**Issue by akarl10** [cockpit-project/cockpit#16083]
Cockpit version: 248
OS: ubuntu impish (developement branch) armhf

this does not happen on x64

Steps to reproduce

1. install and enable pcp metrics
2. wait some time until data should be recorded

the result is that it keeps saying "No data available" even tough pmval -f 3 disk.partitions.write -a /var/log/pcp/pmlogger/... clearly shows that there is data.
see bug 12747

**Comment by martinpitt** [comment]
Internal metrics work fine:

```
(printf '\n{ "command": "open", "channel": "m", "payload": "metrics1", "source": "internal", "interval": 1000, "metrics": [ { "name": "cpu.basic.user", "derive": "rate" } ] }\n---\n'; sleep 5) | cockpit-bridge --interact=---
m
{"timestamp":712685459,"now":712685459,"interval":1000,"metrics":[{"name":"cpu.basic.user","derive":"rate","units":"millisec","semantics":"counter"}]}
---
m
[[false]]
---

{"command":"ready","channel":"m"}

---
m
[[9.9800399201340042]]
---
```

But indeed PCP metrics don't:
```
(printf '\n{ "command": "open", "channel": "m", "payload": "metrics1", "source": "pcp-archive", "interval": 1000, "metrics": [ { "name": "kernel.all.cpu.sys", "derive": "rate" } ] }\n---\n'; sleep 1) | cockpit-bridge --interact=---

{"command":"authorize","challenge":"plain1:6d617274696e:","cookie":"super1"}
---

{"command":"ready","channel":"m"}
---

{"command":"close","channel":"m"}
---
```

On x86, they should show something like
```
---
m
[[],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false],[false]]
---
m
{"timestamp":1628497815951,"now":1628505521506,"interval":1000,"metrics":[{"name":"kernel.all.cpu.sys","derive":"rate","units":"millisec","semantics":"counter"}]}
---
m
[[false],[false],[3],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]]
---
m
[[],[24],[],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[],[],[25],[24],[],[],[]]
```

**Comment by martinpitt** [comment]
Before  I lose it: This [test-pmarchive.c](https://github.com/cockpit-project/cockpit/files/6955353/test-pmarchive.c.txt) program works fine, both with `PM_CONTEXT_LOCAL` and with `PM_CONTEXT_ARCHIVE`. So at least the PCP API itself seems fine, and we somehow use it incorrectly.

The first failed call is the `pmFetch()` call in `on_idle_batch()` (which is what's getting called for pcp-archive). It fails with -12370 aka "End of PCP archive log". So at that point the damage is done. I did not yet see any meaningful difference between what it does on x86 and on arm, but I'll keep digging.

**Comment by martinpitt** [comment]
Fixed in #16220
